chroot_setup.py

```diff
@@ -0,0 +1,373 @@
+  if (CHROOT_DEBUG == False):
+    os.chown(args.chroot_directory + file, user)
+
+#TODO: documentation
+def chroot_chown_dir (user: str , path: str) {
+  for root, dirs, files in os.walk(args.chroot_directory + path):  
+    for dir in dirs:  
+      os.chown(os.path.join(root, dir), user)
+    for file in files:
+      os.chown(os.path.join(root, file), user)
+
+# TODO: documentation
+def chroot_system(line: str):
+  line = line.rstrip()
+  print('system -i "' + line + '"')
+  if (CHROOT_DEBUG == False):
+    os.system('system -i "' + line + '"')
+
+# TODO: documentation
+def chroot_sh(line: str):
+  print(line)
+  if (CHROOT_DEBUG == False):
+    os.system(line)
+
+action_dictionary = {
+  ':file'       : chroot_setup,
+  ':mkidr'      : chroot_mkdir,      # done
+  ':ln_fix_rel' : chroot_ln_fix_rel, # done
+  ':ln_rel'     : chroot_ln_rel,     # done
+  ':ln'         : chroot_ln,         # done
+  ':mknod'      : chroot_mknod,      # done
+  ':cp_dir'     : chroot_cp_dir,     # done
+  ':cp'         : chroot_cp,         # done
+  ':chmod_dir'  : chroot_chmod_dir,  # done
+  ':chmod'      : chroot_chmod,      # done
+  ':chown_dir'  : chroot_chown_dir,  # done
+  ':chown'      : chroot_chown,      # done
+  ':system'     : chroot_system,
+  ':sh'         : chroot_sh
+}
+
+def chroot_setup (chroot_lst):
+  lst_file = open(chroot_lst, 'r')
+  action = None
+
+  # Iterate over all lines in the file, executing actions where encountered
+  for line in lst_file:
+    line = line.strip()
+
+    # Blank line
+    if not line:
+      # print('empty')
+
+    # Comment
+    else if line.startswith('#'):
+      # print('comment')
+
+    # Determine action
+    else if line.startswith(':'):
+      action = action_dictionary[line]
+
+    # Execute action
+    else:
+      if action not None:
+        action(line)
+
+# TODO: documentation
+def initial_check ():
+  if (os.path.exists('/QOpenSys/usr/bin')):
+    CHROOT_DEBUG = False
+    PATH='/QOpenSys/usr/bin:/QOpenSys/usr/sbin:'
+    LIBPATH='/QOpenSys/usr/lib'
+    os.environ['PATH'] = PATH
+    os.environ['LIBPATH'] = LIBPATH
+    print("**********************")
+    print("Live IBM i session (changes made).")
+    print("**********************")
+
+    if (ags.verbose):
+      print('PATH=' + PATH)
+      print('LIBPATH=' + LIBPATH)
+
+  else:
+    CHROOT_DEBUG = True
+    print("**********************")
+    print("Not IBM i, no action is taken (debug flow purpose only).")
+    print("**********************")
+    # exit -1
+
+
+def yum_install(package_list: list):
+  os.environ["LIBPATH"] = '/QOpenSys/pkgs/lib:/QOpenSys/usr/lib'
+  install_template = Template('/QOpenSys/pkgs/bin/yum $yes --installroot=$chroot_dir install $package')
+  auto_yes = '-y' if args.auto_yes else ''
+  for package in package_list:
+    command = install_template.substitute(yes=auto_yes, chroot_dir=args.chroot_directory, package=package)
+    logger.debug(command)
+    os.system(command)
+
+#
+# MAIN
+#
```

Comment:
Mainline code should be under an `if __name__ == '__main__'` block

Reply:
Moved to main block in 10a7597